### DISCUSSION TITLE
Question about restoring official LEGO firmware from SPIKE3

### DISCUSSION THREAD

**Discussion by smolto** [pybricks/support#1267] [General]
I have discovered that when you install SPIKE3 and use the restore to LEGO Legacy option, the reason it does not work correctly is because it is missing practically all the necessary files (_api/, /etc, /sounds etc.) If I make the copy using a tool like rshell and copy all the files works. Is there a way to do it with JavaScript from a web application? Because I am doing it with a script on the different systems but it is a bit complicated.

Files that are necessary
![image](https://github.com/pybricks/support/assets/37777549/17730741-c89b-40f1-9fac-36e8fe8656be)

**Comment by laurensvalk** [comment]
This is the expected result. Just connect to the spike2 or spike3 app, and it will restore all the necessary files automatically.
